src/Collections/Sequences/LittleEndianNatConversions.dfy

```diff
@@ -81,15 +81,38 @@ abstract module {:options "-functionSyntax:4"} LittleEndianNatConversions {
     requires |xs| % E() == 0
     ensures |ys| == |xs| / E()
   {
-    if |xs| == 0 then LemmaDivBasicsAuto(); []
+    if |xs| == 0 then
+      var ys := (LemmaDivBasicsAuto(); []);
+      assert |ys| == |xs| / E();
+      ys
+
     else
       LemmaModIsZero(|xs|, E());
       assert |xs| >= E();
-
       Small.LemmaSeqNatBound(xs[..E()]);
-      LemmaModSubMultiplesVanishAuto();
       LemmaDivMinusOne(|xs|, E());
-      [Small.ToNatRight(xs[..E()]) as Large.uint] + ToLarge(xs[E()..])
+      assert |xs[E()..]| % E() == 0 by {
+        LemmaModSubMultiplesVanishAuto();
+      }
+      var ys := ([Small.ToNatRight(xs[..E()]) as Large.uint] + ToLarge(xs[E()..]));
+      assert |ToLarge(xs[E()..])| == |xs[E()..]| / E();
+      assert |ys| == |xs| / E() by {
+        // To obtain this proof, Mikael Mayer wrote first a detailed proof
+        // Then he removed everything he could that was decreasing the resource count
```

Comment:
> To obtain this proof, Mikael Mayer wrote first a detailed proof

What does that mean? How do I write a detailed proof? Also, I would leave out the name, "To obtain this proof, first write a detailed proof using techniques X, then prune it to reduce the resource count"

Reply:
Oh it's because it's not true in general, with natural integers, that
```
(a + b) / c == a / c + b / c
(a - b) / c == a / c - b / c
```
Proof: Take a == b == 1 and c == 2 for the first (1 == 0), and a == 2, b == 1, c == 2 for the second (0 == 1)

So if you want to make the simplification in any case, proving that b == c is necessary and then you need to invoke the lemma that states how to compute an integer division.